homeassistant/components/switch/mqtt.py

```diff
@@ -31,6 +33,8 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
         _LOGGER.error("Missing required variable: command_topic")
         return False
 
+    payload_on = convert(config.get('payload_on'), str, DEFAULT_PAYLOAD_ON)
```

Comment:
The names are getting very ambiguous. I propose we go for:
 - `command_value_on`
 - `command_value_off`
 - `state_value_on`
 - `state_value_off`

And for backwards compatibility, if `payload_on` or `payload_off` is given, use that value for both.